sdk/core/azure-core/tests/testserver_tests/coretestserver/coretestserver/test_routes/streams.py

```diff
@@ -31,6 +32,16 @@ def streaming_test():
 def stream_compressed_header_error():
     yield b'test'
 
+def stream_compressed_no_header():
+    try:
+        file_path = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "./files/test.tar.gz"))
+        with open(file_path, "rb") as fd:
+            yield fd.read()
+    except:
+        file_path = os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "..", "./files/test.tar.gz"))
+        with open(file_path, "rb") as fd:
+            yield fd.read()
```

Comment:
@marcusc2 Okay I took a closer look at the CI failures and it seems that the issue is still that the test.tar.gz file cannot be found. I think this is arising from the coretestserver being installed on the machine and then CI searching through the installed package path instead of what's here in azure-core (at least that's what it looks like from this path [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1386309&view=logs&j=69247760-d00b-59ba-802c-4e283c193199&t=6654d4af-1717-52b7-cca2-b6c5ec5727ae&l=2605)).

I think an easy way to workaround this (and not have to commit the file to the repo) is to just create this file on the fly. The python standard library includes `gzip` which we can use to create the file, read from it, and then delete it after the test completes. Suggest trying something like this:

```
import gzip
with gzip.open('test.tar.gz', 'wb') as f:
    f.write(b"test")

with open(os.path.join(os.path.abspath('test.tar.gz')), "rb") as fd:
    yield fd.read()

os.remove("test.tar.gz")
```